osu.Game/Beatmaps/Beatmap.cs

```diff
@@ -26,11 +27,26 @@ public double BPMAt(double time)
             return 60000 / BeatLengthAt(time);
         }
 
-        public double BeatLengthAt(double time)
+        public double BPMMultiplierAt(double time)
         {
             ControlPoint overridePoint;
             ControlPoint timingPoint = TimingPointAt(time, out overridePoint);
-            return timingPoint.BeatLength;
+
+            return overridePoint?.VelocityAdjustment ?? 1;
+        }
+
+        public double BeatLengthAt(double time, bool addVelocity = true)

```

Comment:
I don't see how what kind of parameter should be the default is affecting my proposal. Could you elaborate?